src/settings.rs

```diff
@@ -216,7 +243,12 @@ impl Settings {
                 );
 
                 settings.append_to_histfile = add_matches.is_present("append_to_histfile");
-                settings.zsh_extended_history = add_matches.is_present("zsh_extended_history");
+                if add_matches.is_present("zsh_extended_history") {
+                    match settings.history_format {
+                        HistoryFormat::Plain { .. } => settings.history_format = HistoryFormat::Plain { zsh_extended_history: true },
```

Comment:
Yea, this would be cleaner if we just had three history formats, since zsh_extended_history really just implies zsh.